src/hooks/useResponsiveLayout.ts

```diff
@@ -1,27 +1,52 @@
-import {useEffect, useState} from 'react';
-import Navigation from '@libs/Navigation/Navigation';
+import {useEffect, useRef, useState} from 'react';
+import Navigation, {navigationRef} from '@libs/Navigation/Navigation';
 import useWindowDimensions from './useWindowDimensions';
 
 type ResponsiveLayoutResult = {
     shouldUseNarrowLayout: boolean;
     isSmallScreenWidth: boolean;
     isInModal: boolean;
+    isExtraSmallScreenHeight: boolean;
+    isMediumScreenWidth: boolean;
+    isLargeScreenWidth: boolean;
+    isExtraSmallScreenWidth: boolean;
+    isSmallScreen: boolean;
 };
 
 /**
- * Hook to determine if we are on mobile devices or in the Modal Navigator
+ * Hook to determine if we are on mobile devices or in the Modal Navigator.
+ * Use "shouldUseNarrowLayout" for "on mobile or in RHP/LHP", "isSmallScreenWidth" for "on mobile", "isInModal" for "in RHP/LHP".
+ * Note: Don't use "shouldUseNarrowLayout" in popovers and "alert-style" modals.
```

Comment:
Did we consider adding an automatic check for the modal type, such that you can safely use `shouldUseNarrowLayout` in any modal?